src/devtools/client/debugger/src/components/SecondaryPanes/FrameTimeline.tsx

```diff
@@ -261,8 +268,10 @@ function FrameTimeline() {
 }
 
 export default function FrameTimelineSuspenseWrapper() {
+  const selectedFrameId = useAppSelector(getSelectedFrameId);
   return (
     <Suspense
+      key={selectedFrameId ? `${selectedFrameId.pauseId}:${selectedFrameId.frameId}` : null}
```

Comment:
Was this `key` change part a necessary part of the fix? (Can you elaborate on why it was needed, if so?)

Reply:
Yes, this is the core of the fix. We were suspending *within* `FrameTimeline` and we were waiting for the awaited value to resolve there. That never happened because the relevant requested timedout~ (we found a failed assert on BE but no error/timeout was actually returned to FE). So we never had a chance to start listening for a response to a new request - that would get dispatched if we'd rerender with the new source information (`selectedFrameId`). In other words, we were indefinitely waiting for a stale data to be resolved.

Reply:
It surprises me though that you needed to hard-reset the component like this (via `key`)

I would have expected React to bail on the old suspended IO if props/state changed

Reply:
This component has no props and I guess that subscriptions (for state changes) might not even be set up when the component is suspended. So there is no trigger for it to try again in a situation like this.

Reply:
Hmm. It _had_ no props, but now it does (`selectedFrameId `) so...I'm not sure I get it

Reply:
you are right: https://github.com/replayio/devtools/pull/10505